lib/main.js

```diff
@@ -84,7 +89,8 @@ function main(options) {
     let module = JSON.parse(data);
     for (let name in module.dependencies) {
       let values = module.dependencies[name];
-      getInfo(allModules, name).want = getInstallSpec(log, name, values.version, values.from);
```

Comment:
I think if you use `let from = options.from ? values.from : null;` then getInstallSpec() will do the right thing without needing an extra argument. (I.e. it's already doing what you want when `from` argument is falsy.)